pkg/apis/events/v1/conversion.go

```diff
@@ -37,6 +40,12 @@ func Convert_v1_Event_To_core_Event(in *v1.Event, out *k8s_api.Event, s conversi
 	out.FirstTimestamp = in.DeprecatedFirstTimestamp
 	out.LastTimestamp = in.DeprecatedLastTimestamp
 	out.Count = in.DeprecatedCount
+
+	// translate reportingController to out.Source.Component field if it is empty
+	if len(out.Source.Component) == 0 {
+		out.Source.Component = out.ReportingController
+	}
```

Comment:
ugh, this is going to make validation (which runs on internal events) fail here:

https://github.com/kubernetes/kubernetes/blob/5afc42de9501ac7b06bbb349e03a666efd5e3bd7/pkg/apis/core/validation/events.go#L68-L70

Reply:
As we discussed offline, we would revert this conversion